### ISSUE TITLE
Nixpkgs might be affected by "REGULATION OF THE EUROPEAN PARLIAMENT AND OF THE COUNCIL"

### ISSUE THREAD

**Issue by Mic92** [nix-open-org/org-v2#51]
https://eur-lex.europa.eu/legal-content/EN/TXT/?uri=CELEX%3A52022PC0209&qid=1735389305511

### Obligations for software application stores

```
 1.Providers of software application stores shall:

(a)make reasonable efforts to assess, where possible together with the providers of software applications, whether each service offered through the software applications that they intermediate presents a risk of being used for the purpose of the solicitation of children;

(b)take reasonable measures to prevent child users from accessing the software applications in relation to which they have identified a significant risk of use of the service concerned for the purpose of the solicitation of children;

(c)take the necessary age verification and age assessment measures to reliably identify child users on their services, enabling them to take the measures referred to in point (b).

2.In assessing the risk referred to in paragraph 1, the provider shall take into account all the available information, including the results of the risk assessment conducted or updated pursuant to Article 3.

3.Providers of software application stores shall make publicly available information describing the process and criteria used to assess the risk and describing the measures referred to in paragraph 1. That description shall not include information that may reduce the effectiveness of the assessment of those measures.

4.The Commission, in cooperation with Coordinating Authorities and the EU Centre and after having conducted a public consultation, may issue guidelines on the application of paragraphs 1, 2 and 3, having due regard in particular to relevant technological developments and to the manners in which the services covered by those provisions are offered and used. 
```

TL;DR:

We would need to add age verification to NixOS and review our 80K packages if they need to be age restricted. This is a problem because we don't have any accounts, so this is technically not feasible.

**Comment by Mic92** [comment]
The law is currently a draft since 2022. However Germany has a government change next year, which might lead to this law to be approved.

**Comment by Mic92** [comment]
There is currently no exception made for open source. This means the NixOS foundation would be liable and could receive penalty.

**Comment by Mic92** [comment]
@drupol are you not closer to the parliament?

**Comment by drupol** [comment]
No, unfortunately. I'll see if I can get some people to check this out.

**Comment by spiage** [comment]
> ‘software application stores’ means a type of online intermediation services, which is focused on software applications as the intermediated product or service;
> ‘software application’ means any digital product or service that runs on an operating system;

nixpkgs is the packages build/compilation and installation definition _source code_ repository, not the digital 'software application'...
Am I wrong?

**Comment by voronind-com** [comment]
> > ‘software application stores’ means a type of online intermediation services, which is focused on software applications as the intermediated product or service;
> > ‘software application’ means any digital product or service that runs on an operating system;
> 
> nixpkgs is the packages build/compilation and installation definition _source code_ repository, not the digital 'software application'... Am I wrong?

Then the cache is?

**Comment by jtojnar** [comment]
> nixpkgs is the packages build/compilation and installation definition _source code_ repository, not the digital 'software application'...

Many/most of the packages in Nixpkgs are software applications. And Nixpkgs is indeed an online service that intermediates those. So I would not say there is a material difference between Nixpkgs and e.g. Google Play Store according to that definition.

Our best hope would be that this would only target gatekeepers but I do not see that mentioned in CSAR or in the original definition in [Section I, Article 2 of Digital Markets Act](https://eur-lex.europa.eu/legal-content/EN/TXT/?uri=CELEX:52020PC0842).

-----

However, apparently, the proposal has been withdrawn for now: https://en.wikipedia.org/wiki/Regulation_to_Prevent_and_Combat_Child_Sexual_Abuse#Legislative_process

But we will still probably want to keep monitoring and coordinate with other distros when it gets back in discussion.

**Comment by Mic92** [comment]
The issue is that it might be used by lobbyist of proprietary operating system to keep Linux distributions out of public environments such as schools because of missing age filter infrastructure. Even if the laws itself have not been tested in court.
